app/handlers/retrieveResults.py

```diff
@@ -17,5 +22,38 @@ class RetrieveResultsHandler(BaseHandler):
 
     @apiError error_message The error message to display.
     """
+
     def get(self):
-        self.finish("Retrieve Results")
\ No newline at end of file
+        chunk_size = 2048
+        identifier = self.get_body_argument('identifier')
+        project_path = get_project_path(identifier)
+        file_videos = os.path.join(project_path, 'final_videos')
+        file_images = os.path.join(project_path, 'final_images')
+        file_name = os.path.join(project_path, 'results.zip')
+        dir_len = len(project_path)
+
+        zipf = zipfile.ZipFile(file_name, 'w', zipfile.ZIP_DEFLATED)
+        for root, dirs, files in os.walk(file_videos):
+            for file in files:
+                file_path = os.path.join(root, file)
+                zipf.write(file_path, file_path[dir_len :])
+        for root, dirs, files in os.walk(file_images):
+            for file in files:
+                file_path = os.path.join(root, file)
+                zipf.write(file_path, file_path[dir_len :])
+        zipf.close()
+
+        self.set_header('Content-Type', 'application/octet-stream')
+        self.set_header('Content-Description', 'File Transfer')
+        self.set_header('Content-Disposition', 'attachment; filename=' + file_name)
+        with open(file_name, 'rb') as f:
+            try:
+                while True:
```

Comment:
I'm semi worried about infinite loops here (reason being I tried to do a similar thing where I'd break when there were no images in a directory, and that broke). Do you have any ideas about stopping conditions other than `if not data`?